engine/src/main/java/org/destinationsol/IniReader.java

```diff
@@ -67,7 +67,8 @@ public static void write(String fileName, Object... keysVals) {
     }
 
     private void initValueMap(List<String> lines) {
-        for (String line : lines) {
+        for (int lineIndex = 0; lineIndex < lines.size(); lineIndex++) {
```

Comment:
Instead of going back from a for-each iterator to a plain for-statement the in-place modification of `line` should be fixed.

Maybe the following works (the important part is to introduce a new local variable `lineWithoutComment` instead of modifying `line` in-place):

```java
for (String line: lines) {
	String lineWithoutComment = line.split("#", 2)[0];
	String[] sides = lineWithoutComment.split("=");
	...
}
```